src/Api/Response/ResponseParser.php

```diff
@@ -201,27 +205,60 @@ protected function parseProducts(SimpleXMLElement $data)
         return $products;
     }
 
-    /**
-     * @param SimpleXMLElement $data
-     * @return array|null
-     */
-    protected function parseSmartDidYouMean(SimpleXMLElement $data)
+    protected function parseQueryInfoMessage(HttpRequest $request, SimpleXMLElement $data): array
     {
+        if (empty($data->query)) {
+            return [];
+        }
+
         $originalQuery = isset($data->query->originalQuery) ? $data->query->originalQuery->__toString() : null;
         $didYouMeanQuery = isset($data->query->didYouMeanQuery) ? $data->query->didYouMeanQuery->__toString() : null;
         $currentQuery = isset($data->query->queryString) ? $data->query->queryString->__toString() : null;
         $queryStringType = isset($data->query->queryString->attributes()->type)
             ? $data->query->queryString->attributes()->type->__toString()
             : null;
 
-        if ((empty($originalQuery) && empty($didYouMeanQuery)) || $queryStringType === 'forced') {
-            return null;
-        }
+        $requestParams = (array) $request->all();
 
         return [
-            'type' => !empty($didYouMeanQuery) ? 'did-you-mean' : $queryStringType,
-            'alternative_query' => !empty($didYouMeanQuery) ? $didYouMeanQuery : $currentQuery,
-            'original_query' => !empty($didYouMeanQuery) ? $currentQuery : $originalQuery
+            'originalQuery' => $originalQuery,
+            'didYouMeanQuery' => $didYouMeanQuery,
+            'currentQuery' => $currentQuery,
+            'queryStringType' => $queryStringType,
+            'selectedCategoryName' => $this->getSelectedCategoryName($requestParams),
+            'selectedVendorName' => $this->getSelectedVendorName($requestParams)
         ];
     }
+
+    /**
+     * @param array $requestParams
+     * @return string|null
+     */
+    private function getSelectedCategoryName(array $requestParams)
```

Comment:
Add return types.

Reply:
Using `: ?string` throws an error when provisioning plugin on Plentymarkets system. 
`233: UndefError SyntaxError syntax error, unexpected '?'`

Or is there some other way to provide that return type can be null as well, without breaking plugin provisioning?

Reply:
![image](https://user-images.githubusercontent.com/20555112/73341112-463cf780-427c-11ea-8ca0-21a00b8e43bb.png)
.. can't Plentymarkets update their PHP version to something that is actively maintained. PHP 7.0 doesn't even get security fixes anymore:
![image](https://user-images.githubusercontent.com/20555112/73343660-455a9480-4281-11ea-9500-3154e9a9a954.png)

Let's leave out the typehint then.

Reply:
Yup, same reaction :/